ui/app/common/patient-search/models/search.js

```diff
@@ -95,6 +95,12 @@ Bahmni.Common.PatientSearch.Search = function (searchTypes) {
             return patient[column];
         }).join(" - ");
 
+        var abhaIdentifier = null;
+        if (patient.extraIdentifiers) {
+            var objIdentifier = JSON.parse(patient.extraIdentifiers);
+            abhaIdentifier = objIdentifier && objIdentifier["ABHA Address"];
+        }
+        patient.abhaIdentifier = patient.abhaIdentifier ? patient.abhaIdentifier : (abhaIdentifier ? abhaIdentifier : patient.identifier);
```

Comment:
In this case, the variable would be called: `patient.extraIdentifier`

Reply:
Can we pick up first extra identifier from objIdentifier object?

Reply:
We need the name of the Identifier from configuration. Maybe a global property which says.. which extraIdentifier should be searched on? As per my understanding, people may configure multiple identifiers, but want to enable a search only from one of the them (like PAN, or ABHA, or SSN, etc). We should let them choose which one will act as the search identifier (besides the primary one). Sometimes people may want NONE, in which case this configuration will be NULL.

Reply:
This code is in model file - openmrs-module-bahmniapps/ui/app/common/patient-search/models/search.js
       var extraIdentifier = null;
        if (patient.extraIdentifiers) {
            var objIdentifiers = JSON.parse(patient.extraIdentifiers);
            for (var key in objIdentifiers) {
                extraIdentifier = objIdentifiers[key];
                break;
            }
        } else if (patient.extraIdentifierVal) {
            extraIdentifier = patient.extraIdentifierVal;
        }
        patient.extraIdentifier = patient.extraIdentifier ? patient.extraIdentifier : (extraIdentifier ? extraIdentifier : patient.identifier);
It will pick up the first identifier in the extraIdentifiers. If not present then primary identifier will be displayed.
Please advise.